src/smt/smt_engine.cpp

```diff
@@ -1228,6 +1229,8 @@ SmtEngine::~SmtEngine()
     d_definedFunctions->deleteSelf();
     d_fmfRecFunctionsDefined->deleteSelf();
 
+    d_private->unregisterPreprocessingPasses();
```

Comment:
Can we add a comment here that says that this is to make sure that we destroy the passes before we destroy things that they refer to?